### PR TITLE
graph/formats/rdf: add a query language

### PR DESCRIPTION
**Pull request by kortschak**
Please take a look.

### CODE REVIEW

File: graph/formats/rdf/graph.go [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+// existing node ID.
+func (g *Graph) addNode(n graph.Node) {
+	if _, exists := g.nodes[n.ID()]; exists {
+		panic(fmt.Sprintf("rdf: node ID collision: %d", n.ID()))
+	}
+	g.nodes[n.ID()] = n
+	g.ids.Use(n.ID())
+}
+
+// AddStatement adds s to the graph. It panics Term UIDs in the statement
```
Comment (vladimir-ch) [inline graph/formats/rdf/graph.go:51]: Something (an "if"?) is missing after "It panics".

File: graph/formats/rdf/graph_example_test.go [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+	}
+
+	it := g.Nodes()
+	nodes := make([]rdf.Term, 0, it.Len())
+	for it.Next() {
+		nodes = append(nodes, it.Node().(rdf.Term))
+	}
+
+	// Construct a query start point. This can be reused. If a specific
+	// node is already know it can be used to reduce the work required here.
```
[the remark this answers is not in the document]
Comment (kortschak) [reply graph/formats/rdf/graph_example_test.go:130]: Done.

File: graph/formats/rdf/query.go [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+		for it.Next() {
+			if ConnectedByAny(q.g.Edge(it.Node().ID(), s.ID()), fn) {
+				r.terms = append(r.terms, it.Node().(Term))
+			}
+		}
+	}
+	return r
+}
+
+// And returns a query that holds the disjunction of q and p.
```
[the remark this answers is not in the document]
Comment (kortschak) [reply graph/formats/rdf/query.go:66]: Thank you. You've fixed a persistent misapprehension that has lasted decades.

Fixed.

File: graph/formats/rdf/graph.go (hunk 1 above)
Comment (kortschak) [reply graph/formats/rdf/graph.go:51]: Yes. Fixed.

**Review by vladimir-ch — APPROVED** [review]

**Pull request merged**